plugin.rb

```diff
@@ -252,13 +265,19 @@ def after_authenticate(auth, existing_account: nil)
     log("after_authenticate response: \n\ncreds: #{auth['credentials'].to_hash}\nuid: #{auth['uid']}\ninfo: #{auth['info'].to_hash}\nextra: #{auth['extra'].to_hash}")
 
     if SiteSetting.oauth2_fetch_user_details?
+
       if fetched_user_details = fetch_user_details(auth['credentials']['token'], auth['uid'])
         auth['uid'] = fetched_user_details[:user_id] if fetched_user_details[:user_id]
         auth['info']['nickname'] = fetched_user_details[:username] if fetched_user_details[:username]
         auth['info']['image'] = fetched_user_details[:avatar] if fetched_user_details[:avatar]
         ['name', 'email', 'email_verified'].each do |property|
           auth['info'][property] = fetched_user_details[property.to_sym] if fetched_user_details[property.to_sym]
         end
+
+        DiscoursePluginRegistry.custom_user_details.each do |detail|
+          prop = detail.split('.').last.to_sym
```

Comment:
I think this puts us at risk of a clash? e.g. if a plugin did

```
register_custom_user_detail('user.name')
register_custom_user_detail('organisation.name')
```
Both of those would be stored under `name`.

Could we just use the entire path? I think ruby hashes and JSON are both capable of handling periods in object keys. Then later, you'd access it like:

```
user_associated_account.extra['user.name']
```

Reply:
Or alternatively, we could make `register_custom_user_detail` take two parameters: the property name, and the json path?

```
DiscoursePluginRegistry.register_custom_user_detail({
  prop: 'myusername',
  path: 'user.name'
}, plugin)
DiscoursePluginRegistry.register_custom_user_detail({
  prop: 'orgname',
  path: 'organisation.name'
}, plugin)
```

Reply:
I think you can still clash if two different plugins use this register. Using the full path seems safer to me.